### DISCUSSION TITLE
UniFi OS - Dream Machines 4.x.x. on boot scripts vs native firmware

### DISCUSSION THREAD

**Discussion by rjcds** [unifi-utilities/unifi-common#622] [Q&A] [answered]
My understanding is that with 4.x, Ubi finally got around to implementing a lot of features natively that were being filled by boostchicken's excellent utilities

I have stayed on 3.0.2 and use the on boot scripts primarily for persistent ssh keys, and to run peacey's excellent split-vpn scripts for policy based routing

Can anyone on 4.x let me know if the native UnifiOS has these features implemented properly, and if not, whether they have successfully migrated the scripts to 4.x ?

**Comment by rjcds** [comment, marked as answer]
Just in case anyone stumbles across the same question in future - I bit the bullet and upgraded, and both split-vpn and persistent ssh key scripts working without a hitch

**Reply by ddkedr** [reply]
Hey! There is an issue in peacey's repo that says that  split-vpn doesn't work on 4.x UnifiOS. Here is an issue — https://github.com/peacey/split-vpn/issues/211

Can you tell a little more about how you use split-vpn?

I'm still afraid to upgrade from 3.x because of that. Thanks

**Comment by rjcds** [comment]
Hi

I use it to force one specific client on my network to use a wireguard
tunnel, with a killswitch if the tunnel is down

Worked without any adjustment after upgrading from 3.x

I might be possible to do this with native features in 4.x's Network App
<https://help.ui.com/hc/en-us/articles/12566175125783-UniFi-Gateway-Policy-Based-Routes>
but I still haven't tried to use Unifi's WG interface (as I'm using the
command line WG setup from 2.x onwards, and I can't be bothered setting up
new keys etc on my clients as everything still works)
